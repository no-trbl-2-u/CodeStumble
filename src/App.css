.App {
  background-color:rgb(89, 140, 149);
  height: 100%;
  margin:0px;
  padding:0px;
}


/*                        Media Queries                            */

@media screen and (max-width: 475px) {
  .sideNavBlock {
    visibility: hidden;
    width: 0%;
    margin: 0px;
    padding: 0px;
  }

  .float {
    visibility: visible;
  }
}