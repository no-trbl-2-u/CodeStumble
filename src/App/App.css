.App {
  background-color:white;
  height: 100%;
  margin:0px;
  padding:0px;
  font-family: 'Fira Code';
}


/*                        Media Queries                            */

@media screen and (max-device-width: 550px) {
  .sideNavBlock {
    visibility: hidden;
    width: 0%;
    margin: 0px;
    padding: 0px;
  }

  .float {
    visibility: visible;
  }

  .App-header {
    /* visibility: visible; */
    color: red;
  }}

@media screen and (max-width: 750px) and (max-height: 640px) {
  .sideNavBlock {
    visibility: hidden;
    width: 0%;
    margin: 0px;
    padding: 0px;
  }

  .float {
    visibility: visible;
  }

  .App-header {
    visibility: visible;
  }
}