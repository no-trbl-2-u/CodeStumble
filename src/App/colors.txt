#91B974
#F22749
#5C67A7
#2F4F4F
#291F34