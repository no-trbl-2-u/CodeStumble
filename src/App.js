import React, { Component } from 'react';
import { Router, Link } from '@reach/router'

import SideNav from './components/sideNav/SideNav';
import NavButton from './components/NavButton/NavButton'
import Content from './components/content/Content';

import ArticleIndex from './components/content/ArticleIndex/ArticleIndex';
import About from './components/content/About/About.js'
import Article from './components/content/ArticleIndex/Article/Article'
import Home from './components/content/Home/Home'

import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App container-fluid">
        <SideNav/>

        <Content>
          <Router>
            <Home path='/'/>
            <About path='/about' />
            <ArticleIndex path='/article' />
            <Article path='/article/:articleId'/>
          </Router>
        </Content> 

        <NavButton />
      </div>
    );
  }
}

export default App;
