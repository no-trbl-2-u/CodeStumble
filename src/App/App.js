// App.js
import React, { Component } from 'react';
import { Router } from '@reach/router'

import SideNav from '../components/sideNav/SideNav';
import NavButton from '../components/NavButton/NavButton'
import Content from '../components/content/Content';
import ArticleIndex from '../components/content/ArticleIndex/ArticleIndex';
import About from '../components/content/About/About.js'
import Article from '../components/content/Article/Article'
import Home from '../components/content/Home/Home'

import articles from '../components/content/Article/md_articles/index'

// Import the App's css
import './App.css';

class App extends Component {
  constructor () {
    super();
    this.state = {
      articles,
    }
  }

  componentDidMount() {
  }

  render() {
    const {articles} = this.state

    return (
      <div className="App container-fluid">
        <SideNav/>
        {/* By splitting Content, the Router renders
                the child component within Content */}
        <Content>
          <Router>
            <Home path='/'/>
            <About path='/about' />
            <ArticleIndex 
              path='/articles'
              articles={articles} 
            />
            <Article
              path='/articles/:articleId'
              articles={articles}
            />
          </Router>
        </Content> 

        <NavButton />
      </div>
    );
  }
}

export default App;
