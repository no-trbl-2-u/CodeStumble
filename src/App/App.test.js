// App.test.js
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App'
import renderer from 'react-test-renderer'

it(`renders without crashing`, () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
})

// Take a snapshot and compare it
it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
  tree.children.map(ea => console.log(ea.children))
});