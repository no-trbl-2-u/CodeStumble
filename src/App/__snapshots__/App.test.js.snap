// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App container-fluid"
>
  <div
    className="sideNavBlock"
  >
    <h3>
       Code Stumble 
    </h3>
     
    <br />
     
    <br />
    <div
      className="portImgBlock"
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="WebmasterIMG"
          className="portfolioImg img-circle"
          src="Retro landscape.jpg"
        />
      </a>
    </div>
     
    <br />
     
    <br />
     
    <br />
    <div
      className="linkBox"
    >
      <ul>
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          <li>
            <i
              className="fa fa-home my-float"
            />
            <span>
               Home
            </span>
          </li>
        </a>
         
        <br />
        <a
          href="/about"
          onClick={[Function]}
        >
          <li>
            <i
              className="fa fa-user my-float"
            />
            <span>
               About
            </span>
          </li>
        </a>
         
        <br />
        <a
          href="/article"
          onClick={[Function]}
        >
          <li>
            <i
              className="fa fa-list-ul my-float"
            />
            <span>
               Article Index
            </span>
          </li>
           
          <br />
        </a>
      </ul>
    </div>
  </div>
  <div
    className="content-container"
  >
    <header>
      <h2
        className="App-header"
      >
        <i>
          Code Stumble
        </i>
      </h2>
    </header>
    <div
      role="group"
      style={
        Object {
          "outline": "none",
        }
      }
      tabIndex="-1"
    >
      <div
        className="Home-container"
      >
        <h1
          id="welcome_banner"
        >
          Welcome to Code Stumble
        </h1>
        <h3
          align="center"
        >
           If you're not lost, you're not exploring! 
        </h3>
      </div>
    </div>
  </div>
  <div>
    <span
      className="float"
      id="menu-share"
    >
      <i
        aria-hidden="true"
        className="fa fa-diamond my-float"
      />
    </span>
    <ul
      id="navButtonMenu"
    >
      <li>
        <a
          href="/article"
          onClick={[Function]}
        >
          <i
            className="fa fa-list-ul my-float"
          />
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          <i
            className="fa fa-user my-float"
          />
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          <i
            className="fa fa-home my-float"
          />
        </a>
      </li>
    </ul>
    <div
      className="label-container"
    >
      <div
        className="label-text"
      >
         Navigate 
      </div>
      <i
        className="fa fa-play label-arrow"
      >
         
      </i>
    </div>
  </div>
</div>
`;
